wallet-sdk/src/main/kotlin/org/stellar/walletsdk/extension/Server.kt

```diff
@@ -18,14 +21,20 @@ import org.stellar.walletsdk.util.formatAmount
  * @return Account response object
  *
  * @throws [AccountNotFoundException] when account is not found
+ * @throws [HorizonRequestFailedException] for Horizon exceptions
  */
 @Throws(AccountNotFoundException::class)
 suspend fun Server.accountByAddress(accountAddress: String): AccountResponse {
   try {
     return accounts().account(accountAddress)
-  } catch (e: IOException) {
-    // TODO: check that error code is 404
-    throw AccountNotFoundException(accountAddress)
+  } catch (e: Exception) {
+    if (e is ErrorResponse) {
+      if (e.code == 404) {
+        throw AccountNotFoundException(accountAddress)
+      }
+      throw HorizonRequestFailedException(e)
+    }
+    throw e
```

Comment:
We have the same logic repeated lower in the file. Should we create a helper to reuse in both places?

Reply:
Yeah can you make a function 
```kotlin
return safeHorizonCall {
  accounts().account(accountAddress)
}
```